evm_arithmetization/src/generation/mod.rs

```diff
@@ -151,12 +153,33 @@ pub struct TrieInputs {
     pub storage_tries: Vec<(H256, HashedPartialTrie)>,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
+pub struct TrimmedTrieInputs {
+    /// A partial version of the state trie prior to these transactions. It
+    /// should include all nodes that will be accessed by these
+    /// transactions.
+    pub state_trie: HashedPartialTrie,
+    /// A partial version of each storage trie prior to these transactions. It
+    /// should include all storage tries, and nodes therein, that will be
+    /// accessed by these transactions.
+    pub storage_tries: Vec<(H256, HashedPartialTrie)>,
+}
+
+impl TrieInputs {
+    pub(crate) fn trim(&self) -> TrimmedTrieInputs {
+        TrimmedTrieInputs {
+            state_trie: self.state_trie.clone(),
+            storage_tries: self.storage_tries.clone(),
+        }
+    }
+}
```

Comment:
what does this accomplish, that would not be accomplished by using the derived `Clone` trait and `self.tries.clone()` below?

Reply:
We're getting rid of the txn and receipt tries. Similarly for `TrimmedGenerationInputs` where we're getting rid of unneeded fields post initial processing.